src/WebJobs.Script.WebHost/WebScriptHostExceptionHandler.cs

```diff
@@ -0,0 +1,51 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Runtime.ExceptionServices;
+using System.Threading.Tasks;
+using System.Web.Hosting;
+using System.Web.Http;
+using Microsoft.Azure.WebJobs.Host.Timers;
+
+namespace Microsoft.Azure.WebJobs.Script.WebHost
+{
+    public class WebScriptHostExceptionHandler : IWebJobsExceptionHandler
+    {
+        public void Initialize(JobHost host)
+        {
+        }
+
+        public Task OnTimeoutExceptionAsync(ExceptionDispatchInfo exceptionInfo, TimeSpan timeoutGracePeriod)
+        {
+            WebScriptHostManager manager = GetManager();
+            manager?.Instance.TraceWriter.Error("A function timeout has occurred. Host is shutting down.");
+            manager?.Instance.TraceWriter.Flush();
+
+            // Give the manager some time to shut down gracefully.
+            Task.Delay(timeoutGracePeriod).ContinueWith((ct) =>
+            {
+                HostingEnvironment.InitiateShutdown();
+            });
+
+            manager?.Stop();
+
+            return Task.CompletedTask;
+        }
+
+        public Task OnUnhandledExceptionAsync(ExceptionDispatchInfo exceptionInfo)
+        {
+            WebScriptHostManager manager = GetManager();
+            manager?.Instance.TraceWriter.Error("An unhandled exception has occurred. Host is shutting down.", exceptionInfo.SourceException);
+            manager?.Instance.TraceWriter.Flush();
+
+            HostingEnvironment.InitiateShutdown();
+            return Task.CompletedTask;
```

Comment:
I've made a minor improvement. I can always come back later and improve this if we find it necessary.

---
In reply to: [77680364](https://github.com/Azure/azure-webjobs-sdk-script/pull/604#discussion_r77680364) [](ancestors = 77680364)